posthog/models/constance.py

```diff
@@ -0,0 +1,46 @@
+import json
+from contextlib import contextmanager
+from typing import Any
+
+from django.db import models
+
+from posthog.settings import CONSTANCE_CONFIG, CONSTANCE_DATABASE_PREFIX
+
+
+class Constance(models.Model):
+    class Meta:
+        constraints = [models.UniqueConstraint(fields=["key"], name="unique key",)]
+
+    key: models.CharField = models.CharField(max_length=128, null=False, blank=False)
+    raw_value: models.CharField = models.CharField(max_length=1024, null=False, blank=True)
+
+    @property
+    def value(self):
+        return json.loads(self.raw_value)
+
+
+def get_dynamic_setting(key: str) -> Any:
+    assert key in CONSTANCE_CONFIG, f"Unknown dynamic setting: {repr(key)}"
+
+    saved_setting = Constance.objects.filter(key=CONSTANCE_DATABASE_PREFIX + key).first()
+    if saved_setting is not None:
+        return saved_setting.value
+    else:
+        return CONSTANCE_CONFIG[key][0]  # Get the default value
+
+
+def set_dynamic_setting(key: str, value: Any):
+    Constance.objects.update_or_create(
+        key=CONSTANCE_DATABASE_PREFIX + key, defaults={"raw_value": json.dumps(value)},
+    )
+
+
+@contextmanager
+def override_constance_config(key: str, value: Any):
```

Comment:
Actual `constance`'s `override_config()` `kwargs`-based approach was super ergonomic IMO, so if this could handle multiple keys at the same time using that API it'd be great (though not a crucial thing)

Reply:
Agreed - do you consider this a blocker? I discovered this many refactors in, so not looking forward to updating every callsite again.

Reply:
Not blocking as I don't think we need more-than-single overrides right now, but I can see us doing that in the future, which would result in some ugly nesting, at which point a refactor may be useful